account_commission_sales_fee/models/commission_settlement.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2022 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class CommissionSettlement(models.Model):
+    _inherit = "commission.settlement"
+
+    def _prepare_sales_fee_product(self):
+     sales_fee_product = self.env["product.product"].search(
+         [("sales_fee", "=", True)], limit=1
+     )
+     if not sales_fee_product:
+         raise UserError(
+             _(
+                 "You don't have the sales fee product. Please create sales fee product."
+             )
+         )
+     return {
+         "product_id": sales_fee_product.id,
+         "quantity":len(self.line_ids),
+         "price_unit": sales_fee_product.standard_price * -1,
+     }
+    
+    def _prepare_invoice(self, journal, product, date=False):
+        res = super()._prepare_invoice(journal, product, date)
+        if res["invoice_line_ids"]:
+            res["invoice_line_ids"].append((0, 0, self._prepare_sales_fee_product()))
+            if "line_ids" in res:
+                res.pop("line_ids", None)
```

Comment:
@AungKoKoLin1997 Please explain why we should choose to update `invoice_line_ids` over `line_ids`.  Also, MAKE SURE TO ADD COMMENTS when the design rationale is not obvious just by looking at the code.